config/v1/types_infrastructure.go

```diff
@@ -500,6 +505,58 @@ type AWSResourceTag struct {
 	Value string `json:"value"`
 }
 
+// DNSAndLBConfiguration contains information about API and API-Int Load Balancers
+// and who is responsible for that configuration.
+// +union
+type DNSAndLBConfiguration struct {
+	// provider determines which DNS solution is in use for this cluster.
+	// When the user has pre-created the Load Balancers for API and API-Int and the
```

Comment:
I'm concerned this might be ambiguous for things like UPI, would you expect for UPI that this would be UserProvided or ClusterProvided?

Reply:
The original UPI comment was about load balancers, but I think this is still a good question regarding DNS. For UPI, the installer simply doesn't know.